### PR TITLE
Feature/add cobra features

### LINKED ISSUE #30: Enable developpers to use the full potential of cobra
## Description

**As a** `developer`,

**I want** `to be able to use the full potential of cobra (such as prerun function and all)`

**so that** `verdeter could actually become a sort of competitor to cobra`.

Right know, verdeter will work really well if you are not used to cobra. If you are, you will probably miss the flexibility of cobra such as creating aliases, arg validation, prerun function, and more...

### PR DESCRIPTION
**Pull request by slashformotion**
fixes #30

### CODE REVIEW

File: testapp/README.md [hunk 1]
```diff
@@ -4,22 +4,31 @@ Some inputs and associated outputs. When testing modifications, you should have
 
 ```txt
 $ MATH_ORGANISATION_NAME=ditrit go run main.go
+PersistentPreRunE
```
Comment (Zorin95670) [inline testapp/README.md:7]: I don't understand what that could mean.
do you want to see, `PersistentPreRunE` is the log if user used this mecanisme ?
Same for all this file

File: testapp/main.go [hunk 2]
```diff
@@ -5,21 +5,51 @@ import (
...
-	``,
+var myCommand = verdeter.BuildVerdeterCommand(verdeter.VerdeterConfig{
+	Use:         "math",
+	Long:        "math app is a test application for verdeter",
+	Annotations: map[string]string{},
+	PersistentPreRun: func(cmd *cobra.Command, args []string) {
+		println("PersistentPreRun")
+	},
+	PersistentPreRunE: func(cmd *cobra.Command, args []string) error {
+		println("PersistentPreRunE")
```
Comment (Zorin95670) [inline testapp/main.go:20]: use log or whatever but this is not understanble...

**Review by Zorin95670 — CHANGES_REQUESTED** [review]

File: testapp/README.md (hunk 1 above)
Comment (slashformotion) [reply testapp/README.md]: I removed the calls to the pre-run and post-run, it is not really needed. It was meant to show the "new" features.

File: testapp/main.go (hunk 2 above)
Comment (slashformotion) [reply testapp/main.go]: removed

File: testapp/README.md [hunk 3]
```diff
@@ -1,30 +1,25 @@
...
-Some inputs and associated outputs. When testing modifications, you should have this result.
+This golang module is a test app for verdeter.
+This test app use all the major feature of verdeter.
+
+This app and this file will be kept up to date with verdeter development.
+You can see some bash lines and their associated outputs.
 
 ```txt
 $ MATH_ORGANISATION_NAME=ditrit go run main.go
-PersistentPreRunE
```
Comment (Zorin95670) [inline testapp/README.md]: Bad squash commit, still present in the first commit.

**Review by Zorin95670 — CHANGES_REQUESTED** [review]

**Review by Zorin95670 — APPROVED** [review]

**Pull request merged**